string actionName
string[] executableArgNames
string[] paramNames
float64[] paramDefaults
float64[] paramMins
float64[] paramMaxs
